lib/Doctrine/ORM/Internal/Hydration/AbstractHydrator.php

```diff
@@ -397,11 +397,21 @@ protected function hydrateColumnInfo($key)
                 // the current discriminator value must be saved in order to disambiguate fields hydration,
                 // should there be field name collisions
                 if ($classMetadata->parentClasses && isset($this->_rsm->discriminatorColumns[$ownerMap])) {
+                    $discriminatorValues = array_map(
```

Comment:
> Not sure if it's relevant for `v2.x` but for `v3.x` we could create the list of all discriminator values **of the subclasses** during the metadata creation.

More than that, questions to the metadata should be asked directly to the metadata, rather than done in random metadata consumers

Reply:
This block is an `O(1)` called at hydrator initialisation, if I'm not mistaken.

Please move it to a `private` method

Reply:
@lcobucci extremely flaky bench test results, but they seem to stay around normal values, so it is OK IMO